openssl/src/x509/store.rs

```diff
@@ -125,6 +126,19 @@ impl X509StoreBuilderRef {
         unsafe { cvt(ffi::X509_STORE_set_flags(self.as_ptr(), flags.bits())).map(|_| ()) }
     }
 
+    /// Sets the certificate purpose.
+    /// The purpose value can be obtained by `X509Purpose::get_by_sname()`
+    #[corresponds(X509_STORE_set_purpose)]
+    pub fn set_purpose(&mut self, purpose: X509PurposeId) -> Result<(), ErrorStack> {
+        unsafe {
+            cvt(ffi::X509_STORE_set_purpose(
+                self.as_ptr(),
+                purpose.value() as c_int,
```

Comment:
If you have X509PurposeId store a c_int you can avoid this conversion.

Reply:
Done.